test/extended/networking/services.go

```diff
@@ -65,4 +69,93 @@ var _ = Describe("[sig-network] services", func() {
+				expectNoError(err)
+				if len(service.Status.LoadBalancer.Ingress) == 0 {
+					return ""
+				}
+				return service.Status.LoadBalancer.Ingress[0].IP
+			}, 5*time.Minute, 1*time.Second).Should(Equal(""))
+
+			// Test external ip policy configured with allowedCIDRs. Make sure service
+			// is assigned with an ip address that is with allowedCIDRs range
+			By("update network config with allowed cidr for external ip")
+			modifyNetworkConfig(oc, nil, []string{"192.168.132.10/32"}, nil)
+			serviceName = "test-service-with-external-ip"
+			createPodForService(f, serviceName, "")
+			By("check service is within external ip within allowed range")
+			Eventually(func() string {
+				createWebserverLBService(f, serviceName, "", []string{"192.168.132.10"})
+				service, err := f.ClientSet.CoreV1().Services(f.Namespace.Name).Get(context.Background(), serviceName, metav1.GetOptions{})
+				expectNoError(err)
+				var ingressIP string
+				if len(service.Status.LoadBalancer.Ingress) > 0 {
+					ingressIP = service.Status.LoadBalancer.Ingress[0].IP
+				}
+				deleteService(f, serviceName)
+				return ingressIP
+			}, 10*time.Minute, 25*time.Second).Should(Equal("192.168.132.10"))
+			Eventually(func() string {
+				createWebserverLBService(f, serviceName, "", []string{"192.168.132.20"})
+				service, err := f.ClientSet.CoreV1().Services(f.Namespace.Name).Get(context.Background(), serviceName, metav1.GetOptions{})
+				expectNoError(err)
+				var ingressIP string
+				if len(service.Status.LoadBalancer.Ingress) > 0 {
+					ingressIP = service.Status.LoadBalancer.Ingress[0].IP
+				}
+				deleteService(f, serviceName)
+				return ingressIP
+			}, 10*time.Minute, 25*time.Second).Should(Equal(""))
+
+			// Test external ip policy configured with both policy and auto assign cidr. Make sure service
+			// is assigned with an ip address from auto assign cidr.
+			By("update network config with auto assign cidr")
+			modifyNetworkConfig(oc, []string{"192.168.132.254/29"}, []string{"192.168.132.0/29"}, []string{"192.168.132.8/29"})
+			serviceName = "test-service-external-ip-auto-assign"
+			createPodForService(f, serviceName, "")
+			createWebserverLBService(f, serviceName, "", []string{})
+			By("check load balancer service having ingress ip prefix")
+			var ingressIP string
+			Eventually(func() string {
+				service, err := f.ClientSet.CoreV1().Services(f.Namespace.Name).Get(context.Background(), serviceName, metav1.GetOptions{})
+				expectNoError(err)
+				if len(service.Status.LoadBalancer.Ingress) == 0 {
+					return ""
+				}
+				ingressIP = service.Status.LoadBalancer.Ingress[0].IP
+				return ingressIP
+			}, 5*time.Minute, 1*time.Second).Should(HavePrefix("192.168.132"))
+
+			// Revert cluster networks config into default settings and make sure
+			// service must not assigned with ingress ip address.
+			By("update network config without external ip")
```

Comment:
Also, test again that you are not able to create a service with ExternalIP (eventually).